cider/exec/nextgen/jitlib/llvmjit/LLVMJITFunction.cpp

```diff
@@ -161,6 +164,67 @@ JITValuePointer LLVMJITFunction::emitJITFunctionCall(
   }
 }
 
+// look up a runtime function based on the name
+JITValuePointer LLVMJITFunction::emitRuntimeFunctionCall(
+    const std::string& fname,
+    const JITFunctionEmitDescriptor& descriptor) {
+  auto func = module_.module_->getFunction(fname);
+  if (!func) {
+    LOG(FATAL) << "Function: " << fname << " does not exist.";
+  }
+  cloneFunctionRecursive(func);
+
+  llvm::SmallVector<llvm::Value*, JITFunctionEmitDescriptor::DefaultParamsNum> args;
+  args.reserve(descriptor.params_vector.size());
+  for (auto jit_value : descriptor.params_vector) {
+    LLVMJITValue* llvmjit_value = static_cast<LLVMJITValue*>(jit_value);
+    args.push_back(llvmjit_value->llvm_value_);
+  }
+
+  llvm::Value* ans = ir_builder_->CreateCall(func, args);
+  return std::make_unique<LLVMJITValue>(
+      descriptor.ret_type, *this, ans, "ret", JITBackendTag::LLVMJIT, false);
+}
+
+void LLVMJITFunction::cloneFunctionRecursive(llvm::Function* fn) {
```

Comment:
So this indicates a nested function like `substr(lower(col_str))`?

Reply:
Yes, it will copy all dependent functions of the input function.